Algorithms/ShuffleTheArray.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Algorithms
+{
+    public class ShuffleTheArray
+    {
+        /// <summary>
+        /// Complexity: O(n) - For each element from 0 to n, We use flag to determine order of elements that we'll add a new array
+        /// </summary>
+        /// <param name="numsArray">Numbers array</param>
+        /// <param name="n">Items to take and combine</param>
+        /// <returns>Returns the array with combined values</returns>
+        public int[] Shuffle(int[] numsArray, int n)
+        {
+            if (numsArray == null) return new int[] { };
+            if (numsArray.Length == 0) return numsArray;
+            if (n < 1 || n > 500 || ((n * 2) != numsArray.Length)) return new int[] { };
+
+            int lenArray = numsArray.Length;
+            int[] newArr = new int[lenArray];
+            bool flag = true;
```

Comment:
can you think in a better name than `flag`?

Reply:
Sure, i think so.

Reply:
Just want to point this out. After thinking in better naming, you figure it out how to clean your code and ended up with a better version of the solution. That's what you need to do everytime :)

Reply:
I understand, thanks for this advice 👌  👍 . In fact I always try, but sometimes I get stuck 😕